### PR TITLE
[AI]  Played with patch and mock to add some more tests

### PR DESCRIPTION
**Pull request by Grummel7**
...we may as well keep them

### CODE REVIEW

File: default/python/tests/AI/test_system_network.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    # fo interface asks for an empire_id, so we need two arguments as well
+    def getImmediateNeighbors(self, sys_id, _):
+        return TestUniverse.connected_systems.get(sys_id, {})
+
+
+def test_connections():
+    """Test that the graph is bi-directional."""
+    for i1, lanes in TestUniverse.connected_systems.items():
+        for i2 in lanes:
+            assert i1 in TestUniverse.connected_systems[i2]
```
Comment (Cjkjvfnby) [inline default/python/tests/AI/test_system_network.py]: The test that tests the test fixture is a bit overhead. 

Maybe we could just make a function that will get a single-direction graph and return bi-directional?

File: default/python/tests/AI/test_system_network.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+
+def test_get_neighbors():
+    """Rather tests the patch."""
```
Comment (Cjkjvfnby) [inline default/python/tests/AI/test_system_network.py]: I don't thing we need to test a patch itself.

File: default/python/tests/AI/test_system_network.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    with patch("freeOrionAIInterface.getUniverse", new=TestUniverse):
+        assert system_network.get_neighbors(1) == {2, 7}
+
+
+@pytest.mark.parametrize(
+    ("sys_id", "expected"),
+    ((1, {1, 2, 3, 4, 5, 6, 7, 8, 9, 10}), (4, {1, 2, 3, 4, 5, 6, 7, 8, 9, 10}), (12, {11, 12}), (INVALID_ID, {})),
```
Comment (Cjkjvfnby) [inline default/python/tests/AI/test_system_network.py]: Let's format it one case per line.

```suggestion
    (
            (1, {1, 2, 3, 4, 5, 6, 7, 8, 9, 10}),
            (4, {1, 2, 3, 4, 5, 6, 7, 8, 9, 10}),
            (12, {11, 12}),
            (INVALID_ID, {}),
    ),
```

File: default/python/tests/AI/test_system_network.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+)
+def systems_connected_to_system(sys_id, expected):
```
Comment (Cjkjvfnby) [inline default/python/tests/AI/test_system_network.py]: Have you missed the `test_` prefix?

**Review by Cjkjvfnby — COMMENTED** [review]
Pytest also has a way to patch objects. 

By adding this fixture you could remove some patches. 

```python

import freeOrionAIInterface as fo

@pytest.fixture(autouse=True)
def connected_systems(monkeypatch):
    monkeypatch.setattr(fo, "getUniverse", TestUniverse)
```

File: default/python/tests/AI/test_system_network.py (hunk 1 above)
Comment (Grummel7) [reply default/python/tests/AI/test_system_network.py]: I've added a comment why I prefer it this way.

File: default/python/tests/AI/test_system_network.py (hunk 3 above)
Comment (Grummel7) [reply default/python/tests/AI/test_system_network.py]: That's the way I wrote them, but black disagreed with you.
Well, I agree, so I've added another case just to make sure the don't fit into one line :-D

Comment (Cjkjvfnby) [reply default/python/tests/AI/test_system_network.py]: You could use trailing coma to enforce multiline

```python
(1,2,3,)  
```
-> black -> 

```python
(
    1,
    2,
    3,
)
```

**Review by Cjkjvfnby — APPROVED** [review]

File: default/python/tests/AI/test_system_network.py (hunk 3 above)
Comment (Grummel7) [reply default/python/tests/AI/test_system_network.py]: Good to know.

**Pull request merged**